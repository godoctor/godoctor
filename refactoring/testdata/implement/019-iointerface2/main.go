package main // <<<<< stubInterface,1,1,1,1,io.WriterAt,WriteAtStruct,pass

import (
	"fmt"
	"io"
)

func main() {
	fmt.Println("works")

}

type Queue interface {
	Enqueue(x int)
	Dequeue() int
	ThreeQueue(x int) int
}

type Apple struct {
	
}

func (a *Apple) funcWrite (w io.Writer) {
	fmt.Fprintln(w, "Hello")
}
