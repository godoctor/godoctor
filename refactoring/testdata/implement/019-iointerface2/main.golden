package main // <<<<< stubInterface,1,1,1,1,io.WriterAt,WriteAtStruct,pass

import (
	"fmt"
	"io"
)

func main() {
	fmt.Println("works")

}

type Queue interface {
	Enqueue(x int)
	Dequeue() int
	ThreeQueue(x int) int
}

type Apple struct {
	
}

func (a *Apple) funcWrite (w io.Writer) {
	fmt.Fprintln(w, "Hello")
}

type WriteAtStruct struct {
}

func (s *WriteAtStruct) WriteAt(p []byte, off int64) (n int, err error) {
	return 0, nil
}
